app/src/cpp/cfc-cpp/MultiThreadedDecoder.h

```diff
@@ -29,7 +30,10 @@ class MultiThreadedDecoder
 
 	void stop();
 
-	bool legacy_mode() const;
+	int mode() const;
+	bool set_mode(int mode_val);
+	int detected_mode() const;
```

Comment:
This interface is to allow us to switch modes while keeping the MultiThreadedDecoder around when the mode doesn't switch (rather than tearing it down, resetting progress). So we can click the toggle button to our heart's content...